### PR TITLE
Add an exchange refs option to UI compare view

### LINKED ISSUE #5003: Delta table comparison view: exchange directions button
Add an exchange directions button that flips the base and topic delta table refs to compare. 

Screen Shot 2023-01-10 at 15 11 23

### PR DESCRIPTION
**Pull request by talSofer**
Closes #5003

This addition is general to the compare page and not only relevant to delay lake table comparison, therefore is not hidden behind a feature flag. 

Screen Shot 2023-02-07 at 10 22 30

### CODE REVIEW

**Review by johnnyaug — APPROVED** [review]
LGTM - one comment

File: webui/src/pages/repositories/repository/compare.jsx [hunk 1]
```diff
@@ -197,6 +206,20 @@ const CompareList = ({ repo, reference, compareReference, prefix, onSelectRef, o
                         withCommits={true}
                         withWorkspace={false}
                         selectRef={onSelectCompare}/>
+
+                    <OverlayTrigger placement="bottom" overlay={
+                        <Tooltip>Exchange directions</Tooltip>
```
[the remark this answers is not in the document]
Comment (talSofer) [reply webui/src/pages/repositories/repository/compare.jsx]: Thanks, the exact corrections I was looking for :)

File: webui/src/pages/repositories/repository/compare.jsx [hunk 2]
```diff
@@ -197,6 +206,20 @@ const CompareList = ({ repo, reference, compareReference, prefix, onSelectRef, o
                         withCommits={true}
                         withWorkspace={false}
                         selectRef={onSelectCompare}/>
+
+                    <OverlayTrigger placement="bottom" overlay={
+                        <Tooltip>Switch directions</Tooltip>
+                    }>
+                    <span>
+                        <Button variant={"link"} disabled={false}
+                              onClick={e => {
```
Comment (eladlachmi) [inline webui/src/pages/repositories/repository/compare.jsx:215]: Sorry to sound like a broken record here, but handlers shouldn't be anonymous arrow functions and should use `useCallback` to keep the callback function reference stable between renders

File: webui/src/pages/repositories/repository/compare.jsx [hunk 3]
```diff
@@ -45,6 +48,12 @@ const CompareList = ({ repo, reference, compareReference, prefix, onSelectRef, o
         await statistics.postStatsEvents(deltaDiffStatEvents);
     }
 
+    const router = useRouter();
+    const handleExchangeRefs = () => {
```
Comment (eladlachmi) [inline webui/src/pages/repositories/repository/compare.jsx:52]: While we're changing the terminology...
```suggestion
    const handleSwitchRefs = () => {
```
Not a must. Up to you.

File: webui/src/pages/repositories/repository/compare.jsx [hunk 4]
```diff
@@ -197,6 +206,20 @@ const CompareList = ({ repo, reference, compareReference, prefix, onSelectRef, o
                         withCommits={true}
                         withWorkspace={false}
                         selectRef={onSelectCompare}/>
+
+                    <OverlayTrigger placement="bottom" overlay={
+                        <Tooltip>Switch directions</Tooltip>
+                    }>
+                    <span>
+                        <Button variant={"link"} disabled={false}
```
Comment (eladlachmi) [inline webui/src/pages/repositories/repository/compare.jsx:214]: No need to explicitly add `disabled={false}`, since it's the default state

**Review by eladlachmi — CHANGES_REQUESTED** [review]

File: webui/src/pages/repositories/repository/compare.jsx (hunk 2 above)
Comment (talSofer) [reply webui/src/pages/repositories/repository/compare.jsx]: Thanks, I learned something new :)

**Review by Jonathan-Rosenberg — APPROVED** [review]

**Review by eladlachmi — APPROVED** [review]
LGTM

**Pull request merged**